### PR TITLE
Validator - more analytics

### PR DESCRIPTION
**Pull request by shakedbr-codota**
* When closing vscode, validator binary stays alive - FIXED
* Added analytics for validator commands

### CODE REVIEW

File: src/validator/ValidatorClient.ts [hunk 1]
```diff
@@ -206,16 +213,19 @@ export function setIgnore(responseId: string): Promise<string[]> {
...
     const method = "shutdown";
     const body = {
       method: method,
+      params: {},
     };
     const promise = request(body) as Promise<string[]>;
     validationProcess.shutdowned = true;
     return promise;
   }
+  return Promise.resolve();
```
Comment (dimacodota) [inline src/validator/ValidatorClient.ts:228]: why?

**Review by dimacodota — APPROVED** [review]
LGTM, see my question

File: src/validator/ValidatorClient.ts (hunk 1 above)
Comment (shakedbr-codota) [reply src/validator/ValidatorClient.ts:228]: To make the return type constant (Promise<unknown>). Remove this?

Comment (dimacodota) [reply src/validator/ValidatorClient.ts:228]: OK

**Review by dimacodota — APPROVED** [review]
LGTM

**Pull request merged**